### PR TITLE
Send notifications for replies to a users comment

### PR DESCRIPTION
**Pull request by doshitan**
* Utilize the event and queue systems in Laravel 5 to trigger and handle
  notifications for new comments, this requires deployments now include
  a `php artisan queue:listen` or `php artisan queue:work --daemon`
  (each has upsides and downsides) command under some sort of process
  management system (e.g., systemd, supervisor) so that the queuing
  stuff works
* Update references to DBAnnotation model to the Annotation model
* Remove the OldAnnotation model
* Remove soft delete stuff from the Notification model as the required
  columns do not exist in the DB schema (adding the soft delete stuff
  was a mistake made during the Laravel 5 upgrade)

### CODE REVIEW

File: install.md [hunk 1]
```diff
@@ -38,3 +38,6 @@ If you want this to run from the web root instead of http://yoursite.com/madison
 1. Edit your httpd.conf file to change the DocumentRoot to Madison's public directory (e.g. ```DocumentRoot "/var/www/html/madison/public"```)
 2. And allow Madison to handle url rewriting (e.g. ```<Directory "/var/www/html/madison"> AllowOverride All </Directory>```)
 3. Restart Apache
+
+
+You'll need to run `php artisan queue:listen` whenever running the app.
```
Comment (krues8dr) [inline install.md]: Please add a link to the Queue documentation here, to explain what's happening:  https://laravel.com/docs/5.1/queues#running-the-queue-listener

**Comment by krues8dr** [comment]
This is great work @doshitan !  I added one comment to add a bit more documentation, otherwise this is good by me.

**Comment by sethetter** [comment]
Need to add an `ENV` property for `APP_URL` to be used for the `url` setting in `config/app.php` for URL building. Otherwise :+1:

**Comment by doshitan** [comment]
@sethetter good catch, updated, give it another go

**Comment by doshitan** [comment]
Changes needed for each of our Forge deployments:
* [ ] Configure a Queue Worker
* [ ] Set `APP_URL` environment variable to appropriate thing

**Comment by sethetter** [comment]
Looks great! 👍

**Pull request merged**